client/src/main/java/io/apicurio/registry/rest/client/RegistryClient.java

```diff
@@ -131,29 +131,29 @@ default ArtifactSearchResults listArtifactsInGroup(String groupId) {
         return listArtifactsInGroup(groupId, null, null, null, null);
     }
 
-    ArtifactMetaData createArtifact(String groupId, String artifactId, String version, ArtifactType artifactType, IfExists ifExists, Boolean canonical, String artifactName, String artifactDescription, String contentType, InputStream data);
+    ArtifactMetaData createArtifact(String groupId, String artifactId, String version, ArtifactType artifactType, IfExists ifExists, Boolean canonical, String artifactName, String artifactDescription, String contentType, String fromURL, String artifactSHA, InputStream data);
 
-    ArtifactMetaData createArtifact(String groupId, String artifactId, String version, ArtifactType artifactType, IfExists ifExists, Boolean canonical, String artifactName, String artifactDescription, String contentType, InputStream data, List<ArtifactReference> artifactReferences);
+    ArtifactMetaData createArtifact(String groupId, String artifactId, String version, ArtifactType artifactType, IfExists ifExists, Boolean canonical, String artifactName, String artifactDescription, String contentType, String fromURL, String artifactSHA, InputStream data, List<ArtifactReference> artifactReferences);
```

Comment:
Do we also need to allow passing the hashing algorithm here?

Reply:
We can do it, of course, but I'm not sure how much this flexibility will be used in real world ...